schema.c

```diff
@@ -3392,10 +3402,28 @@ _sch_traverse_nodes (sch_instance * instance, sch_node * schema, GNode * parent,
         {
             if (!(flags & SCH_F_FILTER_RDEPTH) || (depth >= rdepth))
             {
-                for (GNode *leaf = child->children; leaf; leaf = leaf->next)
+                /* Access is stored on the * node */
+                sch_node *lschema = sch_node_child_first (schema);
+                if (sch_is_hidden (lschema) ||
+                   (flags & SCH_F_CONFIG && !sch_is_writable (lschema)))
                 {
-                    free (leaf->children->data);
-                    leaf->children->data = g_strdup ("");
+                    DEBUG (flags, "Silently ignoring leaf-list \"%s\"\n", name);
+                    apteryx_free_tree (child);
```

Comment:
when you delete this tree, is it part of a large tree - if so, has child been detached from that tree?

Reply:
apteryx_free_tree calls g_node_destroy on the root (well child in this case) which un-hooks the node from its parent. https://github.com/frida/glib/blob/main/glib/gnode.c#L140

Reply:
OK, that's interesting - I might use that one day.